--- a/Core/AppRuntime/V8Inspector/Source/V8InspectorAgent.cpp
+++ b/Core/AppRuntime/V8Inspector/Source/V8InspectorAgent.cpp
@@ -426,8 +426,8 @@
         }
         v8::Local<v8::String> string_value = v8::Local<v8::String>::Cast(value);
         int len = string_value->Length();
-        std::basic_string<uint16_t> buffer(len, '\0');
-        string_value->Write(v8::Isolate::GetCurrent(), &buffer[0], 0, len);
+        std::vector<uint16_t> buffer(len, 0);
+        string_value->Write(v8::Isolate::GetCurrent(), buffer.data(), 0, len);
         return v8_inspector::StringBuffer::create(
             v8_inspector::StringView(buffer.data(), len));
     }
